### PR TITLE
Add Minitest

### PR DESCRIPTION
**Pull request by blowmage**
This PR converts all remaining RSpec specs to Minitest. Right now the tests are using the Spec DSL to ease the review easier. We can convert the new tests to the assertion style in a future PR.

### CODE REVIEW

File: test/google/gax/error_test.rb [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+# LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+# DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+# THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+require 'test_helper'
+require 'google/gax/errors'
+require 'google/protobuf/any_pb'
+require_relative '../../../spec/fixtures/fixture_pb'
```
Comment (quartzmo) [inline test/google/gax/error_test.rb]: Does this file use types from `fixture_pb`? I don't see any usages. If not, can we remove this line?

File: test/google/gax/util_test.rb [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+# DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+# THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+require 'test_helper'
+require 'google/gax'
+require 'google/protobuf/any_pb'
+require 'google/protobuf/timestamp_pb'
+require_relative '../../../spec/fixtures/fixture_pb'
```
Comment (quartzmo) [inline test/google/gax/util_test.rb]: Can `fixtures/fixture_pb.rb` be copied to the `test` dir in this PR to prepare for deleting `spec` in a follow-up PR?

File: test/google/gax/util_test.rb [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+      _(user).must_be_kind_of(Google::Protobuf::User)
+      _(user.name).must_equal(USER_NAME)
+      _(user.map_field).must_be_kind_of(Google::Protobuf::Map)
+      user.map_field.each do |k, v|
+        _(MAP[k]).must_equal v
+      end
+    end
+
+    it 'handles IO instances' do
+      file = File.new('spec/fixtures/fixture_file.txt')
```
Comment (quartzmo) [inline test/google/gax/util_test.rb]: Can `spec/fixtures/fixture_file.txt` be copied to the test dir in this PR to prepare for deleting spec in a follow-up PR?

**Comment by blowmage** [comment]
Pull and try again. I added a commit that added a `get_application_default` stub that I had removed earlier and I didn't think it was being used/needed.

File: test/google/gax/util_test.rb (hunk 2 above)
Comment (blowmage) [reply test/google/gax/util_test.rb]: Yes, but I didn't want to make any changes to the spec directory in this PR. I planned on moving these files when RSpec is removed.

File: test/google/gax/error_test.rb (hunk 1 above)
Comment (blowmage) [reply test/google/gax/error_test.rb]: It does not. That was ported from the spec file. Removed.

File: test/google/gax/util_test.rb (hunk 2 above)
Comment (quartzmo) [reply test/google/gax/util_test.rb]: Can fixtures/fixture_pb.rb be copied to the test dir in this PR to prepare for deleting spec in a follow-up PR? Copying will not change the spec dir.

**Comment by quartzmo** [comment]
Credentials error fixed, thanks! 👍

File: test/google/gax/util_test.rb (hunk 2 above)
Comment (blowmage) [reply test/google/gax/util_test.rb]: Eh, changed my mind, moved.

File: test/google/gax/util_test.rb (hunk 3 above)
Comment (blowmage) [reply test/google/gax/util_test.rb]: Moved.

**Pull request merged**